framework/quota/test/org/apache/cloudstack/quota/QuotaStatementTest.java

```diff
@@ -0,0 +1,248 @@
+        injectMockToField(configDao, "_configDao");
+        injectMockToField(quotaUsage, "_quotaUsage");
+        injectMockToField(emailQuotaAlert, "_quotaAlert");
+    }
+
+    @Test
+    public void testStartStop() {
+        try {
+            quotaStatement.start(); // expected to fail as pid is not available
+        } catch (NumberFormatException ignored) {
+        }
+        assertTrue(quotaStatement.stop());
+    }
+
+    @Test
+    public void testStatementPeriodBIMONTHLY() {
+        Calendar date = Calendar.getInstance();
+
+        //BIMONTHLY - first statement of month
+        date.set(Calendar.DATE, QuotaStatementImpl.s_LAST_STATEMENT_SENT_DAYS + 1);
+        Calendar period[] = quotaStatement.statementTime(date, STATEMENT_PERIODS.BIMONTHLY);
+        assertTrue(period == null);
+
+        //1 of this month
+        date.set(Calendar.DATE, 1);
+        period = quotaStatement.statementTime(date, STATEMENT_PERIODS.BIMONTHLY);
+        assertTrue(period != null);
+        assertTrue(period[0].toString(), period[0].before(period[1]));
+        assertTrue(period[0].toString(), period[0].get(Calendar.DATE) == 1);
+        assertTrue(period[1].toString(), period[1].get(Calendar.DATE) == 15);
+
+        //BIMONTHLY - second statement of month
+        date = Calendar.getInstance();
+        date.set(Calendar.DATE, QuotaStatementImpl.s_LAST_STATEMENT_SENT_DAYS + 16);
+        period = quotaStatement.statementTime(date, STATEMENT_PERIODS.BIMONTHLY);
+        assertTrue(period == null);
+
+        //17 of this month
+        date.set(Calendar.DATE, 17);
+        period = quotaStatement.statementTime(date, STATEMENT_PERIODS.BIMONTHLY);
+        assertTrue(period != null);
+        assertTrue(period[0].toString(), period[0].before(period[1]));
+        assertTrue(period[0].toString(), period[0].get(Calendar.DATE) == 16);
+
+        //get last day of the previous month
+        Calendar aCalendar = Calendar.getInstance();
+        aCalendar.add(Calendar.MONTH, -1);
+        aCalendar.set(Calendar.DATE, aCalendar.getActualMaximum(Calendar.DAY_OF_MONTH) + 1);
+
+        assertTrue(period[1].toString(), period[1].get(Calendar.DATE) == aCalendar.get(Calendar.DATE));
+
+    }
+
+    @Test
+    public void testStatementPeriodMONTHLY() {
+        Calendar date = Calendar.getInstance();
+        Calendar aCalendar = Calendar.getInstance();
+
+        //MONTHLY
+        date = Calendar.getInstance();
+        date.set(Calendar.DATE, QuotaStatementImpl.s_LAST_STATEMENT_SENT_DAYS + 1);
+        Calendar period[] = quotaStatement.statementTime(date, STATEMENT_PERIODS.MONTHLY);
+        assertTrue(period == null);
+
+        //1 of this month
+        date.set(Calendar.DATE, QuotaStatementImpl.s_LAST_STATEMENT_SENT_DAYS - 1);
+        period = quotaStatement.statementTime(date, STATEMENT_PERIODS.MONTHLY);
+        assertTrue(period != null);
+        assertTrue(period[0].toString(), period[0].before(period[1]));
```

Comment:
Add an assertion that the length of ``period`` is 2 to avoid a confusing ``ArrayIndexOutofBoundsException``.